internal/stategen/cleanup.go

```diff
@@ -238,21 +238,48 @@ func cleanupOwlBotYaml(repoRoot, moduleName string) error {
 		return err
 	}
 
+	// Also check for a cleanup name, which may be different from the module name.
+	// For example, the cloudtasks module has the proto path .../cloud/tasks.
+	// We can derive this name from the postprocessor config.
+	ppc, err := loadPostProcessorConfig(filepath.Join(repoRoot, "internal/postprocessor/config.yaml"))
+	if err != nil {
+		return fmt.Errorf("loading postprocessor config: %w", err)
+	}
+	cleanupName := moduleName // Default to the module name.
+	importPrefix := "cloud.google.com/go/" + moduleName + "/"
+	for _, sc := range ppc.ServiceConfigs {
+		if strings.HasPrefix(sc.ImportPath, importPrefix) {
+			// The InputDirectory is the path to the API protos, relative to the
+			// googleapis root. It has a structure like:
+			// "google/cloud/service/v1" or "google/identity/accesscontextmanager/v1".
+			// The third component of this path is the service name as it appears
```

Comment:
I wouldn't take the third component here - that doesn't work for things like "google/spanner/v1".

But I think the full path *is* always present in the source line, e.g. `source: /google/identity/accesscontextmanager/v1/cloud.google.com/go` (before that was recently deleted). So I'd just use the whole of `sc.InputDirectory` rather than splitting it.